### PR TITLE
Full stack workflow

### PR DESCRIPTION
**Pull request by mo-kalaleeb**
Very Very Fragmented Code, Most of it i left in the controller, Kenan, Please Refactor this into something more structured, Please the code is ugly as F***, This code is not valid for publishing in any means what so ever, have security flaws and structure flaws, i uploaded for everyone to calibrate and fix

### CODE REVIEW

File: backend/config/bundles.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
```
Comment (gh-assan) [inline backend/config/bundles.php:2]: we need to create new bundles : 
1- common => the common code should be here, abstractions like(base controller)
2- painting  => the painting code should be here
3- core        => generic kafka, elastic search should be here

File: backend/features/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+ * Date: 7/21/2019
+ * Time: 12:59 PM
+ */
+
+use Behat\Behat\Context\Context;
+use Symfony\Component\HttpFoundation\Response;
+
+class FeatureContext implements Context
+{
+    private $api_url = "localhost:8000";
```
Comment (gh-assan) [inline backend/features/bootstrap/FeatureContext.php:14]: should be in configuration file

File: backend/features/bootstrap/FeatureContext.php [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+    private $create_endpoint = "/createPainting";
```
Comment (gh-assan) [inline backend/features/bootstrap/FeatureContext.php:16]: each file should know the endpoints to call

File: backend/features/bootstrap/FeatureContext.php [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    // region Painting Create
+
+    /**
+     * @Given /^the backend server is alive$/
+     * @throws Exception
+     */
+
+    public function theBackendServerIsAlive()
+    {
+        $client = new GuzzleHttp\Client(['base_uri' => $this->api_url]);
```
Comment (gh-assan) [inline backend/features/bootstrap/FeatureContext.php:36]: let us abstract the usage of GuzzleHttp\Client
by creating our Client , inside it we can use gazzle

File: backend/features/bootstrap/FeatureContext.php [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+        $client = new GuzzleHttp\Client([
+            'base_uri' => $this->api_url
+        ]);
+
+        $res = $client->request("PUT", $this->update_endpoint, [
+            'json' => [
+                "id" => $paintingID,
+                "name" => "Black Day",
+                "image_url" => "http://localhost:4200/assets/logo.png",
```
Comment (gh-assan) [inline backend/features/bootstrap/FeatureContext.php:166]: ??

File: backend/src/BusinessLogic/CrudI.php [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace App\BusinessLogic;
+
+use Symfony\Component\HttpFoundation\Request;
+
+interface CrudI
```
Comment (gh-assan) [inline backend/src/BusinessLogic/CrudI.php:7]: Use Interface in the name instead of `I`

File: backend/src/BusinessLogic/CrudMysql.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    public function getSetData(Request $request)
+    {
+
+        // Allocate JSON Object
+        $data = json_decode($request->getContent(), true);
+        $request->request->replace(is_array($data) ? $data : array());
+
+        //Get data
```
Comment (gh-assan) [inline backend/src/BusinessLogic/CrudMysql.php:35]: We should call mapper and validator here

File: backend/src/BusinessLogic/CrudMysql.php [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        //Set data
+        $this->painting->setName($name)
+            ->setImageUrl(@$image_url)
+            ->setDescription($description)
+            ->setMedium($medium)
+            ->setCategory($category)
+            ->setSize($size);
+    }
+
+    public function create(Request $request)
```
Comment (gh-assan) [inline backend/src/BusinessLogic/CrudMysql.php:52]: we should call the service here,
before call the mapper and the validator

File: backend/src/BusinessLogic/CrudMysql.php [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        $this->getSetData($request);
+
+        $this->entityManager->persist($this->painting);
+        $this->entityManager->flush();
+    }
+
+    public function update(Request $request)
+    {
+        $id = $request->request->get("id");
```
Comment (gh-assan) [inline backend/src/BusinessLogic/CrudMysql.php:62]: steps should be : 

1- create UpdatePaintingRequest using the mapper
2- validate UpdatePaintingRequest 
3- call PaintingService->update(UpdatePaintingRequest );

in the service : 

1- $painting = $this->paintinManager->get( updatePaintingRequest ->getId());
if empty($painting)  => throw EntityNotFoundException()
else 
 $this->paintinManager->update(updatePaintingRequest );

**Review by mo-kalaleeb — COMMENTED** [review]
This is not valid anymore use backend-v1.0

**Pull request closed without merging**